pyproject.toml

```diff
@@ -104,33 +104,20 @@ structlog = ["structlog"]
 [project.scripts]
 litestar = "litestar.__main__:run_cli"
 
-[tool.hatch.metadata]
-allow-direct-references = true
 
 [tool.hatch.build.targets.sdist]
 include = [
   'docs/PYPI_README.md',
-  '/Makefile',
   '/litestar',
-  '/tests',
-  '/CHANGELOG.rst',
 ]
 
-[tool.pdm]
-ignore_package_warnings = [
-  "alabaster",
-  "sphinxcontrib-*",
-  "sphinx-*",
-  "sphinx",
-  "pre-commit",
-  "pydata-sphinx-*",
-  "slotscheck",
-  "autobahn",
-  "accessible-pygments"
-]
 
-[tool.pdm.dev-dependencies]
+[tool.uv]
+default-groups = ["dev", "linting", "test", "docs"]
+
+[dependency-groups]
 dev = [
+  "litestar[full]",
```

Comment:
```suggestion
dev = [
  "asgi-lifespan",
  "eval-type-backport",
  { include-group = "lintint" },
  { include-group = "doc" },
  { include-group = "test" },
```

Reply:
@provinzkraut It would be good to break the other requirements in to groups as well.  This will make it easier for us to install variants of dev environments with and without the packages included.

Reply:
Which ones specifically? I was looking to port this as close to the PDM one for now, so it's easier to find out why something broke if it does :)

Reply:
Actually, I just tried your suggestion locally and it doesn't work. Without the `litestar[full]`, the extras aren't included by default

Reply:
There's [a comment](https://github.com/astral-sh/uv/issues/8607#issuecomment-2440030178) suggesting that something like `default-extras` might be added, but for now I think we have to do with `litestar[full]`?